python/Ganga/Lib/LCG/Grid.py

```diff
@@ -55,8 +53,15 @@ def __init__(self, middleware='EDG'):
                 'LCG-%s UI has not been configured. The plugin has been disabled.' % self.middleware)
             return
 
-#       create credential for this Grid object
-        self.active = self.check_proxy()
+
+    def get_active(self, create=True):
+        """ This checks if the proxy is valid and returns a True/False
+        If the create=True (default) it will attempt to create a new proxy.
+        If it is ceate=False then no calls which generate a new proxy on disk will be made."""
+        isValid = self.check_proxy(True)
+        return isValid
+    # Replace self.active with call to get_active with default args
+    active = property(get_active)
```

Comment:
I wanted to do this in https://github.com/ganga-devs/ganga/pull/411/files#diff-dcae5641b9610eca0e80152f406817edR2353 but I think it makes more sense to allow to possibly call get_active and check rather than forcibly recreate every single time which this PR likely causes to happen as there is no caching.